### PR TITLE
Minor script and dependency fixes

### LINKED ISSUE #104: ROS packages download failing from TAR
This is the second time I see this happening without changes in our side; in this case with `laser_assembler` and `opencv3`.

`rosinstall_generator` is giving an incorrect output (particularly for the version). The output was correct at some point because the download wasn't failing some days ago, but it's not correct now. For example, `laser_assembler`:
```
juan@Sullust:~$ rosinstall_generator laser_assembler --tar
Using ROS_DISTRO: kinetic
Package names: ['laser_assembler']
From path: None
Deps: False
Deps up to: None
Deps depth: None
Deps only: False
Excludes: None
Exclude paths: None
Wet: True
Flat: False
Tar: True
- tar:
    local-name: laser_assembler
    uri: https://github.com/ros-gbp/laser_assembler-release/archive/release/kinetic/laser_assembler/1.7.4-0.tar.gz
    version: laser_assembler-release-release-kinetic-laser_assembler-1.7.4-0
```
If I download the package at that URI, the folder inside the tar file has a different name than the one specified in the version: `laser_assembler-release-release-kinetic-laser_assembler` (no trailing version number).

I will send a PR correcting this, but it's kinda awful that the contents of a package at a fixed URI change from time to time. @meyerj any thoughts?

### PR DESCRIPTION
**Pull request by jubeira**
Minor fixes:
- `prefix` dir _has_ to be created if it's empty; two prefixes halt the script if it is _not_ empty.
- Closes #104.

### CODE REVIEW

**Review by meyerj — APPROVED** [review]
Just had the same problem. LGTM!

Note that the first run with this patch without cleaning the `catkin_ws` fails with
```
[opencv3] Updating /opt/ros_android/catkin_ws/src/opencv3
Tarball Client does not support updating with different version 'opencv3-release-release-kinetic-opencv3' != 'opencv3-release-release-kinetic-opencv3-3.3.1-5'
[laser_assembler] Updating /opt/ros_android/catkin_ws/src/laser_assembler
Tarball Client does not support updating with different version 'laser_assembler-release-release-kinetic-laser_assembler' != 'laser_assembler-release-release-kinetic-laser_assembler-1.7.4-0'

Exception caught during install: Error processing 'laser_assembler' : [laser_assembler] Update Failed of /opt/ros_android/catkin_ws/src/laser_assembler
Error processing 'opencv3' : [opencv3] Update Failed of /opt/ros_android/catkin_ws/src/opencv3

ERROR in config: Error processing 'laser_assembler' : [laser_assembler] Update Failed of /opt/ros_android/catkin_ws/src/laser_assembler
Error processing 'opencv3' : [opencv3] Update Failed of /opt/ros_android/catkin_ws/src/opencv3
```

File: install.sh [hunk 1]
```diff
@@ -75,10 +75,10 @@ do
             shift
         ;;
         *)
-            if [[ ! -z "$prefix" ]]; then
+            if [[ -z "$prefix" ]]; then
                 if [ ! -d "$1" ]; then
                     set +e
-                    mkdir -p "$1" || help=1; break
```
Comment (ivanpauno) [inline install.sh:81]: The "|| help=1; break" is usefull in the case of an unrecognized option. For example:
./install -oo output, will faill with a message
`mkdir: invalid option -- 'o'
Try 'mkdir --help' for more information.`

Maybe adding an extra case "-*)" is better for catching non-recognized options.

**Review by ivanpauno — APPROVED** [review]
I left a minor comment, but LGTM

File: install.sh (hunk 1 above)
Comment (jubeira) [reply install.sh:81]: > Maybe adding an extra case "-*)" is better for catching non-recognized options.

Good idea; the problem that I had is that using just the prefix path triggered the help message. I'll add the extra case.

**Pull request merged**